demo/admin/vite.config.mts

```diff
@@ -0,0 +1,85 @@
+import react from "@vitejs/plugin-react-swc";
+import { resolve } from "path";
+import { defineConfig } from "vite";
+import { createHtmlPlugin } from "vite-plugin-html";
+
+import { environment as envVarsToLoad } from "./src/environment";
+
+// https://vitejs.dev/config/
+export default defineConfig(({ mode }) => {
+    const plugins = [
+        react({
+            jsxImportSource: "@emotion/react",
+            plugins: [
+                [
+                    "@swc/plugin-emotion",
+                    {
+                        importMap: {
+                            "@mui/system": {
+                                styled: {
+                                    canonicalImport: ["@emotion/styled", "default"],
+                                    styledBaseImport: ["@mui/material", "styled"],
+                                },
+                            },
+                            "@mui/material/styles": {
+                                styled: {
+                                    canonicalImport: ["@emotion/styled", "default"],
+                                    styledBaseImport: ["@mui/material/styles", "styled"],
+                                },
+                            },
+                        },
+                    },
+                ],
+            ],
+        }),
+        createHtmlPlugin({
+            minify: true,
+            entry: resolve(__dirname, "src/loader.ts"),
+            inject: {
+                data: {
+                    environmentValues: envVarsToLoad.map((envVarKey) => ({
+                        key: envVarKey,
+                        value: mode === "production" ? `$${envVarKey}` : process.env[envVarKey],
+                    })),
+                },
+            },
+        }),
+    ];
+   
+    return {
+        plugins,
+        server: {
+            port: Number(process.env.ADMIN_PORT),
+        },
+        define: {
+            // define NODE_ENV for packages using it
+            "process.env.NODE_ENV": mode === "production" ? "'production'" : "'development'",
+        },
+        optimizeDeps: {
+            esbuildOptions: {
+                // Node.js global to browser globalThis. https://github.com/vitejs/vite/discussions/5912
+                // "gloabl is not defined" occures dirctly after loading. used by draft-is package
+                define: {
+                    global: "globalThis",
+                },
+            },
+            include: [
```

Comment:
do you know/understand why this is needed?

Reply:
This is due to our monorepo setup. Vite automatically detects packages in a monorepo and treats it as source code (= no pre-bundling):

> In a monorepo setup, a dependency may be a linked package from the same repo. Vite automatically detects dependencies that are not resolved from node_modules and treats the linked dep as source code. It will not attempt to bundle the linked dep, and will analyze the linked dep's dependency list instead.

However, since were using CommonJS and not ESM, we need to use `include` to pre-bundle our packages:

> However, this requires the linked dep to be exported as ESM. If not, you can add the dependency to [optimizeDeps.include](https://vitejs.dev/config/dep-optimization-options.html#optimizedeps-include) and [build.commonjsOptions.include](https://vitejs.dev/config/build-options.html#build-commonjsoptions) in your config.

This is also the reason why we need to restart the dev server:

> When making changes to the linked dep, restart the dev server with the --force command line option for the changes to take effect.

https://vitejs.dev/guide/dep-pre-bundling.html#monorepos-and-linked-dependencies

Should I add a comment to the option?

Reply:
Known issue: The `vite build` command doesn't work. I've tried to include the Admin packages in `build.commonjsOptions.include` as described in the Vite docs, but that didn't work. Can we ignore that for now?